### PR TITLE
perf: reduce the time taken to pull out your own receipts

### PR DESCRIPTION
**Pull request by kegsay**
By doing a single bulk query rather than one-per-room.

### CODE REVIEW

File: state/receipt_table.go [hunk 1]
```diff
@@ -97,20 +97,30 @@ func (t *ReceiptTable) SelectReceiptsForEvents(roomID string, eventIDs []string)
 }
 
 // Select all (including private) receipts for this user in this room.
```
Comment (DMRobertson) [inline state/receipt_table.go:99]:
```suggestion
// Select all (including private) receipts for this user in these rooms.
```

File: sync3/extensions/receipts.go [hunk 2]
```diff
@@ -95,18 +98,29 @@ func (r *ReceiptsRequest) ProcessInitial(ctx context.Context, res *Response, ext
...
+	for roomID, ownRecs := range ownReceipts {
+		otherReceipts[roomID] = append(otherReceipts[roomID], ownRecs...)
+	}
+
+	for roomID, receipts := range otherReceipts {
+		if len(receipts) == 0 {
 			continue
 		}
-		rooms[roomID], _ = state.PackReceiptsIntoEDU(append(receipts, ownReceipts...))
+		rooms[roomID], _ = state.PackReceiptsIntoEDU(append(receipts, receipts...))
```
Comment (DMRobertson) [inline sync3/extensions/receipts.go:121]: This doesn't look right. Should this be `state.PackReceiptsIntoEDU(receipts)`?

Comment (kegsay) [reply sync3/extensions/receipts.go:121]: Yes, I forgot this bit. This would've duplicated receipts which is why tests were still happy.

Comment (kegsay) [reply sync3/extensions/receipts.go:121]:
```suggestion
		rooms[roomID], _ = state.PackReceiptsIntoEDU(receipts)
```

**Review by DMRobertson — APPROVED** [review]

**Pull request merged**